doc/release/compatibility.rst

```diff
@@ -0,0 +1,150 @@
+
+*   Pre-releases and releases of consequent series are compatible by data
+    layout, binary protocol, and replication protocol.
+
+*   No guarantees are given regarding compatibility between
+    pre-releases/releases of non-consequent release series if the opposite
+    is not stated in the :ref:`release notes <release_notes>`.
+
+*   No guarantees are given regarding compatibility between alpha/beta
+    versions and between alpha/beta and pre-release/release even within one series.
+
+..  _cg_data_layout:
+
+Binary data layout
+------------------
+
+Any newer release (its runtime) is backward compatible with any older one.
+It means the more recent release can work on top of data
+(``*.xlog``, ``*.snap``, ``*.vylog``, ``*.run``) from the older one.
+All functionality of the older release can work in this configuration.
+The same compatibility is maintained between :term:`release series` as well.
+
+An attempt to use a new feature results in one of the options:
+
+*   The attempt is successful.
+
+*   There is a meaningful error about the old data layout until the database schema upgrade.
+    It does not lead to a service outage or data corruption.
+    An instance can upgrade the data layout using the :ref:`box.schema.upgrade() <admin-upgrades>` call
+    to enable all new release features (when all instances of the replicaset are run on the same Tarantool version).
+
+..  _cg_binary_protocol:
+
+Binary protocol
+---------------
+
+The binary protocol evolves without breaking compatibility with old clients.
+All binary protocol requests operational in an older release should work on a newer one.
+They do not change the meaning.
+Responses have the same format, but mappings may contain fields not present in the older release.
+
+``net.box`` client of the older release is able to work
+with the newer one, except the features introduced in the newer release.
+``net.box`` client of the newer release is fully operational with the server
+running under the older one, except the features not implemented in the older release.
+
+..  _cg_replication_protocol:
+
+Replication protocol
+--------------------
+
+An instance run on a newer release may work as:
+
+*   upstream (master) of an instance with an older release
+
+*   downstream (replica) without a database schema upgrade.
+
+The database schema upgrade (``box.schema.upgrade()``) must be performed when all replicaset instances
+run on the same Tarantool version.
+The upgrade does not cause downtime if the application does not lean on internal schema representation.
+
+..  _cg_lua_code:
+
+Lua code
+--------
+
+If a code is run on an older release, it will operate with the same effect on a
+newer one. However, only meaningful code counts.
+If any code throws an error but starts doing something useful, the change is considered compatible.
+
+A room for new functionality is still here: adding new options, more
+fields to a returning table, and more returning values (multireturn).
```

Comment:
```suggestion
fields to a returning table, and more returning values (multireturn).
```
"Returning table" looks like a term here. I haven't seen this term before. What does it mean? Might it be "return table" and "return values"? Please correct me if I'm wrong.